docs/BENCHMARK_GUIDE.md

```
@@ -0,0 +1,398 @@
+criterion_group!(benches, bench_my_feature);
+criterion_main!(benches);
+```
+
+**Key Points:**
+- Use `black_box()` to prevent compiler from optimizing away benchmarked code
+- Test multiple input sizes (at least 5-6 points) to validate complexity claims
+- Set `Throughput` to get per-operation metrics
+- Keep setup outside the timing closure
+
+### 2. Register in Cargo.toml
+
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)